src/shared/machine-id-setup.c

```diff
@@ -30,7 +30,8 @@
 #include "umask-util.h"
 #include "virt.h"
 
-static int acquire_machine_id_from_credential(sd_id128_t *ret) {
+static int acquire_machine_id_from_credential(sd_id128_t *ret_machine_id, bool *ret_machine_id_from_firmware) {
```

Comment:
How would I trigger this though?
`                } else if (IN_SET(detect_vm(), VIRTUALIZATION_KVM, VIRTUALIZATION_AMAZON, VIRTUALIZATION_QEMU, VIRTUALIZATION_XEN) || machine_id_from_firmware || machine_id_from_firmware_credentials) {
`

Since here I need to know if "firmware" was set inside the credential, not only if one was found or not

Reply:
not sure i grok the question? just return > 0 under the condition that you care about.